### PR TITLE
Replace coveralls with codecov

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by certik** [comment]
This looks like the right change.

File: .travis.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ env:
 
   ## Out of tree builds (default):
   # Debug build (with BFD) with LATEST_GCC
-  - BUILD_TYPE="Debug" WITH_BFD="yes" WITH_LATEST_GCC="yes"
+  - BUILD_TYPE="Debug" WITH_BFD="yes" WITH_LATEST_GCC="yes" WITH_COVERAGE="yes"
```
Comment (isuruf) [inline .travis.yml]: This is left from a test run. I'll remove this

**Comment by Sumith1896** [comment]
I don't like this comments :smile:

**Comment by certik** [comment]
Yes, we want to turn those comments off.

**Comment by isuruf** [comment]
I've disabled the comments. Let's see if it is really turned off in another PR

**Pull request merged**